nemo/lightning/resume.py

```diff
@@ -1,183 +1,209 @@
-                another non-NeMo checkpoint format. If import_path is provided, other arguments
-                are unused.
-            adapter_path (str): Path to any adapter checkpoints.
-            resume_if_exists (bool): Whether this experiment is resuming from a previous run. If
-                True, it sets trainer._checkpoint_connector._ckpt_path so that the trainer should
-                auto-resume. exp_manager will move files under log_dir to log_dir/run_{int}.
-                Defaults to False.
-            resume_past_end (bool): By default, AutoResume throws an error if resume_if_exists is
-                True and a checkpoint matching ``*end.ckpt`` indicating a previous training run
-                fully completed. Setting resume_past_end=True disables this behavior and loads the
-                last checkpoint.
-            resume_ignore_no_checkpoint (bool): AutoResume throws an error if resume_if_exists is
-                True and no checkpoint could be found. Setting resume_ignore_no_checkpoint=True
-                disables this behavior, in which case exp_manager will print a message and
-                continue without restoring.
-        """
-        if path and import_path:
-            raise ValueError("Only one of path or import_path can be set")
-
-        self.path = path
-        self.dirpath = dirpath
-        self.import_path = import_path
-        self.adapter_path = adapter_path
-        self.resume_if_exists = resume_if_exists
-        self.resume_past_end = resume_past_end
-        self.resume_ignore_no_checkpoint = resume_ignore_no_checkpoint
-
-    def nemo_path(self, model=None) -> Optional[Path]:
+        trainer_ckpt_path = self.get_trainer_ckpt_path(model)
+        if trainer_ckpt_path:
+            trainer.ckpt_path = trainer_ckpt_path
+            trainer.checkpoint_callback.last_model_path = trainer_ckpt_path
+        elif self.selective_restore_config:
+            new_path = self._try_import_model(
+                model=model,
+                path=self.selective_restore_config.path,
+                adapter_path=self.selective_restore_config.adapter_path,
+            )
+            if isinstance(new_path, AdapterPath):
+                self.selective_restore_config.path = new_path.base_model_path
+                self.selective_restore_config.adapter_path = str(new_path)
+            else:
+                self.selective_restore_config.path = str(new_path)
+            trainer.strategy.selective_restore_config = self.selective_restore_config
+
+    def _try_import_model(
+        self, model: Optional[io.ConnectorMixin], path: str, adapter_path: Optional[str] = None
+    ) -> BasePath:
+        if model is None:
+            raise ValueError("Model is needed to import checkpoint from HF or other non-NeMo checkpoint format.")
+        try:
+            new_path = model.import_ckpt(path)
+        except (ValueError, AttributeError):
```

Comment:
what triggers the ValueError / AttributeError here?

Reply:
It's for when the connector is not found for the particular path.

Reply:
@hemildesai thanks, can you add include your GH comment as a code comment?